oxidation/crates/parsec_client_types/src/local_manifest.rs

```diff
@@ -82,14 +82,14 @@ impl Chunk {
         }
     }
     pub fn from_block_access(block_access: BlockAccess) -> Result<Self, &'static str> {
-        let raw_size = NonZeroU64::try_from(block_access.size).map_err(|_| "invalid raw size")?;
-        let stop = NonZeroU64::try_from(block_access.offset + block_access.size).unwrap();
         Ok(Self {
             id: ChunkID::from(*block_access.id),
             raw_offset: block_access.offset,
-            raw_size,
+            raw_size: block_access.size,
             start: block_access.offset,
-            stop,
+            stop: (block_access.offset + u64::from(block_access.size))
+                .try_into()
+                .unwrap(),
```

Comment:
`from_block_access` return a `Result`, so better replace this unwrap by a proper error handling

Reply:
> It can't fail ! (>= 0) + (> 0) = (> 0)

if it can't fail, we should use the `unwrap_or(|| unreachable!())` syntax and, more importantly, explain why the unwrap is ok in this case

> And if you want to handle overflow, we should change all addition maybe ?

That's a valid point, in the current context the need of handling overflow is really obvious given `block_access.offset/size` can come third party (hence a malicious user can create a block_access.size/offset with `std::u64::MAX`, then all the Parsec client reading those data would crash !)

Can you do a check on the rest of the Rust codebase to see where overflow may occur ? (typically open another PR with `// TODO: what to do with overflow` comments where we need to think about)

Reply:
![image](https://user-images.githubusercontent.com/3187637/163017832-71f9c2b7-a52e-4415-a828-3ab1f335e76b.png)